tests/utilities/test_data.py

```diff
@@ -19,3 +22,51 @@ def test_extract_batch_size():
 
     batch = {"test": [{"test": [torch.zeros(11, 10)]}]}
     assert extract_batch_size(batch) == 11
+
+
+def test_has_iterable_dataset():
+    class MockIterableDataset(IterableDataset):
+        def __iter__(self):
+            yield 1
+            return self
+
+    assert has_iterable_dataset(DataLoader(MockIterableDataset()))
+
+    assert not has_iterable_dataset(DataLoader(Dataset()))
+
+    class MockDataset(Dataset):
+        def __iter__(self):
+            yield 1
+            return self
+
+    assert not has_iterable_dataset(DataLoader(MockDataset()))
+
+
+def test_has_len():
+    class MockDatasetWithLen(Dataset):
+        def __len__(self):
+            return 1
+
+    assert has_len(DataLoader(MockDatasetWithLen()))
+
+    class MockDatasetWithZeroLen(Dataset):
+        def __len__(self):
+            return 0
+
+    with pytest.raises(ValueError):
+        assert has_len(DataLoader(MockDatasetWithZeroLen()))
+
+    assert not has_len(DataLoader(IterableDataset()))
+
+
+def test_get_len():
+    class MockDatasetWithLen(Dataset):
+        def __len__(self):
+            return 1
+
+    assert get_len(DataLoader(MockDatasetWithLen())) == 1
+
+    value = get_len(DataLoader(IterableDataset()))
+
+    assert isinstance(value, float)
+    assert value == float("inf")
```

Comment:
you can use these datasets from the test helpers for dataset vs iterable datasets: https://github.com/PyTorchLightning/pytorch-lightning/blob/e54180363663b690367f0051484650fcd12e5de2/tests/helpers/boring_model.py#L36-L69

the deepsource linter is also warning that `Method '__getitem__' is abstract in class 'Dataset' but is not overridden`: https://deepsource.io/gh/PyTorchLightning/pytorch-lightning/run/5241bb94-206a-464b-8a2b-b2d14afa08c0/python/PYL-W0223 

using the above utiities should fix them. if there are missing variants, you can add to the test helpers too